conans/client/graph/graph_binaries.py

```diff
@@ -283,16 +283,7 @@ def _compute_package_id(self, node, default_package_id_mode, default_python_requ
             # Make sure not duplicated
             indirect_reqs.difference_update(direct_reqs)
         else:
-            node.id_direct_prefs = set()  # of PackageReference
-            node.id_indirect_prefs = set()  # of PackageReference, avoid duplicates
-            for neighbor in neighbors:
-                node.id_direct_prefs.add(neighbor.pref)
-                node.id_indirect_prefs.update(neighbor.id_direct_prefs)
-                node.id_indirect_prefs.update(neighbor.id_indirect_prefs)
-            # Make sure not duplicated, totally necessary
-            node.id_indirect_prefs.difference_update(node.id_direct_prefs)
-            direct_reqs = node.id_direct_prefs
-            indirect_reqs = node.id_indirect_prefs
+            direct_reqs, indirect_reqs = node.package_id_transitive_reqs()
```

Comment:
Probably if we are moving only one branch this function doesn't belong to `node`, but to `self`, or we should move both branches.

Reply:
I put it in the ``node`` because now it needs to be called from 2 different locations: ``graph_binaries.py`` and ``installer.py``. I could leave it here in the ``graph_binaries.py`` as a public static method that receives a single ``node`` as argument, no problem.